lib/teleterm/services/userpreferences/userpreferences.go

```diff
@@ -23,43 +23,95 @@ import (
 	api "github.com/gravitational/teleport/gen/proto/go/teleport/lib/teleterm/v1"
 )
 
-func Get(ctx context.Context, client Client) (*api.UserPreferences, error) {
-	preferences, err := client.GetUserPreferences(ctx, &userpreferencesv1.GetUserPreferencesRequest{})
+func Get(ctx context.Context, rootClient Client, leafClient Client) (*api.UserPreferences, error) {
+	rootPreferencesResponse, err := rootClient.GetUserPreferences(ctx, &userpreferencesv1.GetUserPreferencesRequest{})
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	rootPreferences := rootPreferencesResponse.GetPreferences()
+
+	var leafPreferences *userpreferencesv1.UserPreferences
+	if leafClient != nil {
+		preferences, err := leafClient.GetUserPreferences(ctx, &userpreferencesv1.GetUserPreferencesRequest{})
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		leafPreferences = preferences.GetPreferences()
+	}
+
+	clusterPreferences := rootPreferences.GetClusterPreferences()
+	if leafPreferences != nil {
+		clusterPreferences = leafPreferences.GetClusterPreferences()
+	}
+
 	return &api.UserPreferences{
-		ClusterPreferences:         preferences.GetPreferences().ClusterPreferences,
-		UnifiedResourcePreferences: preferences.GetPreferences().UnifiedResourcePreferences,
+		UnifiedResourcePreferences: rootPreferences.UnifiedResourcePreferences,
+		ClusterPreferences:         clusterPreferences,
 	}, nil
 }
 
 // Update updates the preferences for a given user.
 // Only the properties that are set (cluster_preferences, unified_resource_preferences) will be updated.
-func Update(ctx context.Context, client Client, newPreferences *api.UserPreferences) error {
+func Update(ctx context.Context, rootClient Client, leafClient Client, newPreferences *api.UserPreferences) (*api.UserPreferences, error) {
```

Comment:
Could you later expand the tests for `Update` so that they cover the case with updating leaf preferences?